sdks/python/apache_beam/ml/inference/base.py

```diff
@@ -1416,8 +1498,10 @@ def load():
       if isinstance(side_input_model_path, str):
         self._model_handler.update_model_path(side_input_model_path)
       else:
-        self._model_handler.update_model_paths(
-            self._model, side_input_model_path)
+        if self._model is not None:
```

Comment:
We initially assign it to None, and IIRC there's some update paths where this can get called before it has been assigned. Right now `update_model_paths` is expected to handle (and no-op) the `None` case, so this is at least a bit cleaner

Reply:
How about we:

1)  create a _ModelWrapper or base class, that is used for a single model single process scenario
2) _CrossProcessModelWrapper that also supports multiple models that are shared across the process.
3) Implement  `_ModelWrapper._update_model_path(mh, new_model_path_config)` in both classes inside using the appropriate logic.

that should push some the complexity associated with # of models and  the difference between  update_model_path vs  update_model_paths away from load().

Reply:
just noticed that updating code is wrapped inside another `load()` helper function so maybe that suggestion won't work since at that time we don't have the wrapper yet